src/Features/Core/Portable/ConflictMarkerResolution/AbstractConflictMarkerCodeFixProvider.cs

```diff
@@ -45,91 +45,171 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+                // conflict trivia on the <<<<<<< line (since we would have already reported the
+                // issue there.
+                if (startTrivia.RawKind == _syntaxKinds.ConflictMarkerTrivia)
+                    return false;
+            }
+            else if (position == endLine.Start)
+            {
+                // we were on the >>>>>>> lines.  We only want to report here if there was no
+                // conflict trivia on the ======= or <<<<<<< line (since we would have already reported the
+                // issue there.
+                if (startTrivia.RawKind == _syntaxKinds.ConflictMarkerTrivia ||
+                    middleTrivia.RawKind == _syntaxKinds.ConflictMarkerTrivia)
+                    return false;
+            }
+
+            return true;
+        }
+
+        private bool TryGetConflictLines(
+            SourceText text, int position,
+            out TextLine startLine, out TextLine middleLine, out TextLine endLine)
+        {
+            startLine = default;
+            middleLine = default;
+            endLine = default;
+
+            var lines = text.Lines;
+            switch (text[position])
+            {
+                case '<':
+                    startLine = lines.GetLineFromPosition(position);
+                    return TryFindLineForwards(startLine, '=', out middleLine) &&
+                           TryFindLineForwards(middleLine, '>', out endLine);
+                case '=':
+                    middleLine = lines.GetLineFromPosition(position);
+                    return TryFindLineBackwards(middleLine, '<', out startLine) &&
+                           TryFindLineForwards(middleLine, '>', out endLine);
+                case '>':
+                    endLine = lines.GetLineFromPosition(position);
+                    return TryFindLineBackwards(endLine, '=', out middleLine) &&
+                           TryFindLineBackwards(middleLine, '<', out startLine);
+                default:
+                    throw ExceptionUtilities.UnexpectedValue(text[position]);
+            }
+        }
+
+        private bool TryFindLineForwards(TextLine startLine, char ch, out TextLine foundLine)
+        {
+            var text = startLine.Text!;
+            var lines = text.Lines;
+            for (var i = startLine.LineNumber + 1; i < lines.Count; i++)
+            {
+                var currentLine = lines[i];
+                if (IsConflictMarker(currentLine, ch))
+                {
+                    foundLine = currentLine;
+                    return true;
+                }
+            }
 
-            while (true)
+            foundLine = default;
+            return false;
+        }
+
+        private bool TryFindLineBackwards(TextLine startLine, char ch, out TextLine foundLine)
+        {
+            var text = startLine.Text!;
+            var lines = text.Lines;
+            for (var i = startLine.LineNumber - 1; i >= 0; i--)
```

Comment:
Similarly, startLine can be the first line too.

Reply:
i'm not sure what this means.

Reply:
in that case, the for loop will bail out immediatley.

Reply:
i will add tests :)

Reply:
added.